sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -203,6 +203,70 @@ case class GpuTimeSub(start: Expression,
   }
 }
 
+case class GpuDateAddInterval(start: Expression,
+    interval: Expression,
+    timeZoneId: Option[String] = None)
```

Comment:
DateAdd doesn't take timezone.

It also doesn't support intervals in Spark. How are your tests working I wonder. For reference look at this https://github.com/apache/spark/blob/cde697a479a2f67c6bc4281f39a1ab2ff6a9d17d/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L232

Reply:
I think you got confused between `DateAdd` and `DateAddInterval` (almost same names :) ) . We already support `DateAdd`. This PR is to add support for `DateAddInterval` 

https://github.com/apache/spark/blob/cde697a479a2f67c6bc4281f39a1ab2ff6a9d17d/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L1310

Reply:
oops! sorry I got mixed up